azurelinuxagent/common/event.py

```diff
@@ -631,25 +673,19 @@ def collect_events(self): # pylint: disable=R0914
                         else:
                             self._update_legacy_agent_event(event, event_file_creation_time)
 
-                    event_list.events.append(event)
+                    enqueue_event(event)
                 finally:
                     os.remove(event_file_path)
-            except UnicodeError as e: # pylint: disable=C0103
-                unicode_error_count += 1
-                if len(unicode_errors) < max_collect_errors_to_report:
-                    unicode_errors.append(ustr(e))
-            except Exception as e: # pylint: disable=C0103
-                collect_event_error_count += 1
-                if len(collect_event_errors) < max_collect_errors_to_report:
-                    collect_event_errors.append(ustr(e))
-
-        EventLogger.report_dropped_events_error(collect_event_error_count, collect_event_errors,
-                                                WALAEventOperation.CollectEventErrors, max_collect_errors_to_report)
-        EventLogger.report_dropped_events_error(unicode_error_count, unicode_errors,
-                                                WALAEventOperation.CollectEventUnicodeErrors,
-                                                max_collect_errors_to_report)
-
-        return event_list
+            except ServiceStoppedError as stopped_error:
+                logger.error(
+                    "Unable to enqueue events as service stopped: {0}, skipping events collection".format(
+                        ustr(stopped_error)))
+            except UnicodeError as uni_err:
```

Comment:
are we capturing enough information to identify the specific events that triggered these errors?

Reply:
Umm no, we're not capturing any data about the specific events that can cause the errors, we're mostly capturing the data on what failed.
For extension events though, the errors are logged based on per extension basis, so we atleast know which extension added the bad event (this is only for collecting, not sending telemetry)

Reply:
For extension events though, the errors are logged based on per extension basis,

That would be for extensions using the new pipeline, but not for those using the current pipeline.

Reply:
>That would be for extensions using the new pipeline, but not for those using the current pipeline.

Fair enough, though capturing data would be hard. Like for example if the file contains bad_json we wont be able to parse it, and the only metadata we can capture would be the name. But that name wont tell us enough (its something like 123456.tld). 

What meta-data do you think would be useful here? I can try to capture as much as I can.